### PR TITLE
Add more Windows service info to flare

### PR DESCRIPTION
**Pull request by mrafi97**
### What does this PR do?
This PR changes the function `getServiceStatus` in `/pkg/flare/` to create a file named `servicestatus.json`  that includes details about the Datadog Agent services running in the host. 

The function now gets more details about the Agent services such as the PID, Trigger count ([MSDN](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_trigger_info)), Service Failure Actions ([MSDN](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_failure_actionsa)) and Config ([MSDN](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-query_service_configa)). 

### Motivation
Jira: https://datadoghq.atlassian.net/browse/WINA-81 

### Describe how to test/QA your changes
Install the Agent and create a flare. Check the flare .zip file for `servicestatus.json`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by clarkb7 — APPROVED** [review]

File: pkg/flare/service_windows.go
Comment (sgnn7) [inline pkg/flare/service_windows.go:1]: I would assume that this file's ownership should be @DataDog/windows-agent. I'm guessing maybe even `pkg/flare/archive_win.go`?

File: pkg/flare/service_windows.go
Comment (sgnn7) [inline pkg/flare/service_windows.go:72]: Why not return early in case of errors?

**Review by sgnn7 — COMMENTED** [review]
@mrafi97 Couple of comments

File: pkg/flare/service_windows.go
Comment (mrafi97) [reply pkg/flare/service_windows.go:1]: Hi @sgnn7 yes these files would fall under ownership of @DataDog/windows-agent .

File: pkg/flare/service_windows.go
Comment (mrafi97) [reply pkg/flare/service_windows.go:72]: The idea here would be to continue with trying to attempt to populate the other fields in the struct, `serviceFailureActions`. Since this entire code is composed of a struct with multiple fields that are collected through different Windows API calls, we don't want the function to stop collecting the other information due an error collecting a single field.

Comment (mrafi97) [reply pkg/flare/service_windows.go:72]: It's also why each time an error is returned we're appending it to `err` to which is returned at the end of the function.

File: pkg/flare/service_windows.go
Comment (mrafi97) [reply pkg/flare/service_windows.go:1]: I've added @DataDog/windows-agent as the code owners for these files

File: pkg/flare/service_windows.go
Comment (iglendd) [inline pkg/flare/service_windows.go:95]: Why we just not copy return to sfa.RecoveryActions? What is the point of dividing original Delay?

File: pkg/flare/archive_win.go
Comment (iglendd) [inline pkg/flare/archive_win.go:187]: I feel that a bulk of this function should be moved into new module already dedicated to both flare and services

File: pkg/flare/service_windows.go
Comment (mrafi97) [reply pkg/flare/service_windows.go:95]: Hi @iglendd , 

With `s.RecoveryActions()` the following struct is returned: https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-sc_action

The original struct from Microsoft has a numerical value for `SC_ACTION_TYPE` which needed to be converted to a string that would be more readable for a person viewing this information. Which is why we needed to parse through `recoveryActionSlice` to convert that number to a string.

For some reason the GO windows service manger package multiplies the value for Delay returned by Microsoft by `time.Milliseconds`: https://cs.opensource.google/go/x/sys/+/refs/tags/v0.12.0:windows/svc/mgr/recovery.go;l=81

I had to reverse that because Microsoft already returns the Delay in milliseconds so the multiplication was unnecessary on GO's part.

Comment (iglendd) [reply pkg/flare/service_windows.go:95]: Interesting, thank you makes sense.

File: pkg/flare/service_windows.go
Comment (iglendd) [reply pkg/flare/service_windows.go:72]: It is cleave approach. I like it. I do not think it worth to change, However it is opportunity to express my $0.02. Intuitively at least, and I could be wrong. It is unlikely that one of the API fails and other succeed. But immediate return per @sgnn7 makes code much simpler. In such rare cases if it is indeed possible we would lose some potentially possible information but for the arguably price of more convoluted and  complex code.

File: pkg/flare/archive_win.go
Comment (mrafi97) [reply pkg/flare/archive_win.go:187]: Hey @iglendd , could you elaborate on moving the function to a separate module. Do you mean possibly moving most of the function into another function in `service_windows.go`?

Comment (iglendd) [reply pkg/flare/archive_win.go:187]: Yes, @clarkb7 WDYT?

**Review by iglendd — APPROVED** [review]

File: pkg/flare/archive_win.go
Comment (clarkb7) [reply pkg/flare/archive_win.go:187]: right now I don't see a benefit to having it in one place over another. But with these things it's usually faster to move it than it is to spend time trying to justify it one way or the other.

File: pkg/flare/service_windows.go
Comment (sgnn7) [reply pkg/flare/service_windows.go:72]: I'll defer the final call to @Datadog/windows-agent so all good from my side but my 2c would be that we handle this in a more Golang idiomatic manner (maybe individual methods, a loop, and/or return of multiple error messages 🤷 ) since we only seem to display the last error message in the logs and have a different set of data collected.

**Review by sgnn7 — APPROVED** [review]

File: pkg/flare/service_windows.go
Comment (clarkb7) [reply pkg/flare/service_windows.go:72]: > since we only seem to display the last error message

the errors are accumulated are they not?

Comment (mrafi97) [reply pkg/flare/service_windows.go]: I handled the errors with `log.Warnf` to be more consistent and for simplicity.

File: pkg/flare/archive_win.go
Comment (mrafi97) [reply pkg/flare/archive_win.go:186]: I created a new function `getDDServices` that does the service information collection after connecting to the SC Manager.

**Review by clarkb7 — APPROVED** [review]

**Pull request merged**
